crates/api/src/community/add_mod.rs

```diff
@@ -39,6 +41,10 @@ impl Perform for AddModToCommunity {
       return Err(LemmyError::from_message("not_a_moderator"));
     }
 
+    let new_mod = LocalUserView::read_person(context.pool(), data.person_id).await?;
+    let local_site = LocalSite::read(context.pool()).await?;
+    check_user_approved(&new_mod, &local_site)?;
```

Comment:
Create another helper version of `check_user_approved` that passes in `(context.pool(), person_id)` . 

I see a few places where you could move the fetching into that helper function and cut down on code.